### PR TITLE
Hide `EntityAutocompletePicker` only if there are zero available options

### PR DESCRIPTION
**Pull request by zhammer**
## Hey, I just made a Pull Request!

I noticed some odd behavior in the `EntityAutocompletePicker`. If there is one available option, the picker will be hidden. This can make sense in *some* cases, but in others be quite confusing.

Let's say I have a new catalog and have just begun using tags on my entities. Right now I only use one tag, "critical", to distinguish entities in a critical path. 5 of my 20 components have `tags: ['critical']`. _The tag picker won't render on the catalog page._ As a user I expect that I can filter by my one "critical" tag.

It seems like `EntityAutocompletePicker` evolved from the `EntityNamespacePicker` (see: https://github.com/davidan01/backstage/commit/9b3ab37e1e9b75a9b68e82c92e6c7456ea1dce32), which would explain this: `metadata.namespace` is a required field; if there's only one available option, that means that there's "nothing to pick from" (every entity has the same value, `default`).

On string fields like namespace, one option would be to have `/api/catalog/entity-facets?facet=metadata.namespace` return the number of entities where the value is unset. Then the picker could distinguish between these two cases: 

```json
{
    "facets": {
        "metadata.namespace": [
            {
                "value": "default",
                "count": 2325
            },
            {
                "value": null,
                "count": 0
            },
        ]
    }
}
```

```json
{
    "facets": {
        "metadata.labels.language": [
            {
                "value": "typescript",
                "count": 8
            },
            {
                "value": null,
                "count": 4
            },
        ]
    }
}
```

This would be a *bit* trickier with tags though, as the facets endpoint returns counts of individual tags, vs. the entire tag list on an entity.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by zhammer** [comment]
No rush on this. Curious about people's thoughts. This'd affect the namespace picker too, as that'd show `default` where it's currently hidden.

**Review by tudi2d — CHANGES_REQUESTED** [review]
Hey Zach - thanks for you contribution! This totaly make sense as it could look to end users like it might be deactivated for other reasons than just containing a single option.

**Comment by tudi2d** [comment]
Ah & DCO needs to be fixed.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

As mentioned we should be able to just drop the line in the component implementation.

Needs DCO before we're able to merge

File: plugins/catalog-react/src/components/EntityAutocompletePicker/EntityAutocompletePicker.test.tsx [hunk 1]
```diff
@@ -88,6 +88,56 @@ describe('<EntityAutocompletePicker/>', () => {
...
   });
 
+  it('hides filter if there are no available options', async () => {
+    const mockCatalogApiNoOptions: Partial<jest.Mocked<CatalogApi>> = {
+      getEntityFacets: jest.fn().mockResolvedValue({
+        facets: {
+          'spec.options': [],
+        },
+      }),
+    };
```
[the remark this answers is not in the document]
Comment (zhammer) [reply plugins/catalog-react/src/components/EntityAutocompletePicker/EntityAutocompletePicker.test.tsx]: what do you think of 44652dea070e

File: plugins/catalog-react/src/components/EntityAutocompletePicker/EntityAutocompletePicker.test.tsx [hunk 2]
```diff
@@ -46,20 +46,25 @@ class EntityOptionFilter implements EntityFilter {
   }
 }
 
-describe('<EntityAutocompletePicker/>', () => {
-  const mockCatalogApi: Partial<jest.Mocked<CatalogApi>> = {
+const makeMockCatalogApi = (
+  opts: string[] = defaultOptions,
+): Partial<jest.Mocked<CatalogApi>> => {
+  return {
```
Comment (tudi2d) [inline plugins/catalog-react/src/components/EntityAutocompletePicker/EntityAutocompletePicker.test.tsx:52]: looks good to me - hope this is a bit cleaner:)

**Review by tudi2d — APPROVED** [review]
looks good - let's get this in:)

**Comment by zhammer** [comment]
no stale! merge! lol

**Review by Rugvip — APPROVED** [review]
Thank you! 🎉 

Let's :shipit:

**Pull request merged**